components/server/src/user/user-service.ts

```diff
@@ -421,10 +420,22 @@ export class UserService {
         // update user
         user.name = user.name || authUser.name || authUser.primaryEmail;
         user.avatarUrl = user.avatarUrl || authUser.avatarUrl;
-
+        await this.onAfterUserLoad(user);
         await this.updateUserIdentity(user, candidate, token);
     }
 
+    async onAfterUserLoad(user: User): Promise<User> {
+        let result = user;
+        // migrate user to team only attribution
+        const shouldMigrate = this.configCatClientFactory().getValueAsync("team_only_attribution", false, {
+            user,
+        });
+        if (User.is(result) && (await this.migrationService.needsMigration(result)) && (await shouldMigrate)) {
+            result = await this.migrationService.migrateUser(result);
+        }
+        return result;
+    }
```

Comment:
If the migration fails, we'd fail this request which ultimately fails the check for terms acceptance. This has the potential to prevent the user from using gitpod, if the migration fails. Do we need to catch the error here (and increment a metric/log?)